mirar/processors/database/database_inserter.py

```diff
@@ -87,8 +87,12 @@ def _apply_to_sources(self, batch: SourceBatch) -> SourceBatch:
 
                 assert len(res) == 1
 
-            for key, val in primary_key_dict.items():
-                source_table[key] = val
+                primary_key_df = pd.concat([primary_key_df, res], axis=0).reset_index(
```

Comment:
Don't do this. Make a list and then generate a dataframe at the end. Iterative concat is not considered good practise for pandas I think.